### PR TITLE
191-unsubscribe-from-events

### LINKED ISSUE #191: As an attendee, I want to be able to unsubscribe myself from an event, so that is does not show in the list of attending events.

### PR DESCRIPTION
**Pull request by ros-dorian**

### CODE REVIEW

File: app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingListFragment.java [hunk 1]
```diff
@@ -95,6 +106,32 @@ public void onActivityCreated(@Nullable Bundle savedInstanceState) {
...
+    private void removeEvent(Context context, int position) {
+        new AlertDialog.Builder(context)
+                .setTitle("Leave event")
+                .setMessage("Are you sure you want to unsubscribe from this event?")
+                .setPositiveButton(android.R.string.yes, (dialog, which) -> {
+                    DatabaseObject<Event> event = mAdapter.get(position);
+                    mViewModel.leaveEvent(event.getId());
+                })
+                .setNegativeButton(android.R.string.no, ((dialog, which) -> {
+                    mAdapter.notifyItemChanged(position);
```
Comment (CorentinBT) [inline app/src/main/java/ch/epfl/sdp/ui/main/attending/AttendingListFragment.java]: Useless curlybrace

File: app/src/main/java/ch/epfl/sdp/ui/SwipeToDeleteCallback.java [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+    }
+
+    @Override
+    public void onChildDraw(@NonNull Canvas c, @NonNull RecyclerView recyclerView, @NonNull RecyclerView.ViewHolder viewHolder, float dX, float dY, int actionState, boolean isCurrentlyActive) {
+        super.onChildDraw(c, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive);
+
+        View itemView = viewHolder.itemView;
+        int itemHeight = itemView.getHeight();
+
+        boolean isCancelled = dX == 0 && !isCurrentlyActive;
```
Comment (CorentinBT) [inline app/src/main/java/ch/epfl/sdp/ui/SwipeToDeleteCallback.java]: Pourquoi pas mettre les conditions direct dans le `if`?

Comment (ros-dorian) [reply app/src/main/java/ch/epfl/sdp/ui/SwipeToDeleteCallback.java]: Juste

**Review by CorentinBT — APPROVED** [review]

**Review by ksabaa15 — APPROVED** [review]
Not much to say, the code looks fine and it works manually. Well done.

**Pull request merged**